cli/bpmetadata/path_test.go

```diff
@@ -0,0 +1,157 @@
+package bpmetadata
+
+import (
+	"github.com/stretchr/testify/assert"
+	"path"
+	"regexp"
+	"testing"
+)
+
+const (
+	bptestdataPath = "../testdata/bpmetadata"
+)
+
+func TestIcon(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		want    bool
+		wantErr bool
+	}{
+		{
+			name:    "valid",
+			path:    "assets/icon.png",
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name:    "invalid",
+			path:    "assets/icon2.png",
+			wantErr: true,
+		},
+		{
+			name:    "empty",
+			path:    "",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := isPathValid(path.Join(bptestdataPath, tt.path))
+			if (err != nil) != tt.wantErr {
+				t.Errorf("isPathValid() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if got != tt.want {
+				t.Errorf("isPathValid() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestDirContent(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		regex   string
+		want    []BlueprintMiscContent
+		wantErr bool
+	}{
+		{
+			name:  "valid examples",
+			path:  "content/examples",
+			regex: regexEx,
+			want: []BlueprintMiscContent{
+				BlueprintMiscContent{
+					Name:     "terraform",
+					Location: "examples/acm/acm-terraform-blog-part1/terraform",
```

Comment:
Nested examples are tricky. Here the name should probably be `acm-terraform-blog-part1` rather than `terraform`. I am leaning towards just surfacing top level for now (so from [gke](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/examples/acm-terraform-blog-part1) example, this would be `name:acm-terraform-blog-part1, path:examples/acm-terraform-blog-part1`) and figure out if we should just unnest nested examples (I think these should be few)

Reply:
This is a little more nuanced. See [examples](https://github.com/terraform-google-modules/terraform-google-vm/tree/master/examples/compute_instance) for terraform-google-vm. Here we have nested folders with meaning names as opposed to how examples for done for GKE.

IMO, we should keep the nested folder name since this ACM example is an outlier for a myriad of reasons. I've created a [bug](http://b/253292517) to discuss.

Reply:
Good point, let's discuss next week.

Reply:
Fair enough. Let's discuss and we can change accordingly. b/254079604